src/controller/CHIPDeviceController.cpp

```diff
@@ -1125,10 +1124,9 @@ CHIP_ERROR DeviceCommissioner::SendOperationalCertificateSigningRequestCommand(D
     Callback::Cancelable * successCallback = mOpCSRResponseCallback.Cancel();
     Callback::Cancelable * failureCallback = mOnCSRFailureCallback.Cancel();
 
-    uint8_t CSRNonce[kOpCSRNonceLength];
-    ReturnErrorOnFailure(Crypto::DRBG_get_bytes(CSRNonce, sizeof(CSRNonce)));
-
-    ReturnErrorOnFailure(cluster.OpCSRRequest(successCallback, failureCallback, ByteSpan(CSRNonce, sizeof(CSRNonce))));
+    ByteSpan nonce;
+    ReturnErrorOnFailure(device->GenerateCSRNonce(nonce));
+    ReturnErrorOnFailure(cluster.OpCSRRequest(successCallback, failureCallback, nonce));
```

Comment:
```cpp
ReturnErrorOnFailure(device->GenerateCSRNonce());
ReturnErrorOnFailure(cluster.OpCSRRequest(successCallback, failureCallback, device->GetCSRNonce());
```

seems better since it shows that csr nonce storage is owned by the device.